iocore/net/SSLClientUtils.cc

```diff
@@ -86,6 +93,10 @@ verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
       }
       SSLDebug("IP verification failed for (%s)", buff);
     }
+    if (netvc->getClientVerifyEnable() == 2) {
+      Warning("Server certificate verification failed but continuing with the connection establishment");
+      return preverify_ok;
+    }
```

Comment:
I think you want to add the server name in the message to give actionable data to the operations folks.   That way they can hunt down the badly behaving origins.